bin/common.sh

```diff
@@ -6,49 +6,87 @@ readonly clustersFile="$ENV_DIR/env/clusters.yaml"
 readonly helmfileOutputHide="(^\W+$|skipping|basePath=|Decrypting)"
 readonly helmfileOutputHideTpl="(^[\W^-]+$|skipping|basePath=|Decrypting)"
 readonly replacePathsPattern="s@../env@${ENV_DIR}@g"
+readonly otomiToolsImage="otomi/tools:1.4.10"
 
-get_k8s_version() {
+has_docker=$( docker --version &>/dev/null && echo "1" || echo "0")
+dind=0
+cmd_image=''
+
+function print_error() {
+    echo "Error: " $@ >&2
+}
+
+function _shadow() {
+  local cmd="$1"
+  shift
+  local args="$@"
+  if [[ $has_docker -eq 1 && ("${IN_DOCKER}" != "1" || $dind -eq 1) ]]; then
+    docker run --rm \
+      -v ${ENV_DIR}:${ENV_DIR} \
+      -e CLOUD="$CLOUD" \
+      -e IN_DOCKER="1" \
+      -e CLUSTER="$CLUSTER" \
+      ${otomiToolsImage} ${cmd} ${args}
+  elif command -v ${cmd} &>/dev/null; then
+    command ${cmd} ${args}
+  else
+    print_error "Docker is not available and ${cmd} is not installed locally"
+    exit 1
+  fi
+}
+
+function yq() {
+  local yq_args="$@"
+  _shadow yq "${yq_args}"
```

Comment:
When tested with quoting, it worked well. But I can change it to `_rind yq $@` if preferred.

Reply:
Yes, let's be precise since we know exactly where we want to invoke dockerized version and also where not (in the `bin/*.sh` scripts). So we don't have to use a "smart" function that has to figure this out anymore.